src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1787,6 +1787,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     void ControlCore::ClearSearch()
     {
         const auto lock = _terminal->LockForWriting();
+
+        // GH #19358: select the focused search result before clearing search
+        if (const auto focusedSearchResult = _terminal->GetSearchHighlightFocused())
+        {
+            _terminal->SetSelectionAnchor(focusedSearchResult->start);
+            _terminal->SetSelectionEnd(focusedSearchResult->end);
+            _renderer->TriggerSelection();
+        }

```

Comment:
Personally, I think leaving behind selected text after performing a search feels _a little_ weird, but I think it's necessary. This enables the user to begin a selection at the focused search result using _only_ the keyboard. If we get rid of this, the user would need to manually click on the terminal or transfer focus over somehow.